CMakeLists.txt

```diff
@@ -57,6 +57,11 @@ set(HOLOHUB_DATA_DIR "${CMAKE_BINARY_DIR}/data" CACHE PATH "Data Download direct
 set(CMAKE_INSTALL_RPATH
     "\$ORIGIN:\$ORIGIN/../../../lib:\$ORIGIN/../../lib:/opt/nvidia/holoscan/lib/")
 
+# Set the default CUDA target platforms
```

Comment:
Should add the Holoscan SDK https://github.com/nvidia-holoscan/holoscan-sdk/blob/main/cmake/modules/SetupCUDA.cmake to Holohub and use that?
This would make sure that Holohub and Holoscan SDK use the same mechanism.

I also noticed that there are may instances of CUDA_ARCHITECTURES and CMAKE_CUDA_ARCHITECTURES scattered throughout all the CMakeLists.txt files. Should we remove them and consolidate in this one central location?

Except for the tool_tracking_postprocess which needs `native` because it's using dynamic parallelism.

Reply:
Great suggestion. I'll update this PR to align to `SetupCUDA.cmake`.

The existing values for `CUDA_ARCHITECTURES` / `CMAKE_CUDA_ARCHITECTURES` throughout HoloHub generally look ok. We should allow projects to override the HoloHub default. I'd lean towards leaving those as-is, but probably no major impact either way (except for the tool tracking case you noted).

Reply:
I'm reconsidering whether we should use `SetupCUDA.cmake` to set HoloHub project defaults due to the inlined  [enable_language call](https://github.com/nvidia-holoscan/holoscan-sdk/blob/main/cmake/modules/SetupCUDA.cmake#L33). `SetupCUDA.cmake` assumes the Holoscan SDK build environment is in use, whereas HoloHub projects may not require CUDA.

Here are two relevant HoloHub details:
1. The CUDA language should be optional on a per-project basis. Many projects do not depend directly on the CUDA Toolkit nor require CUDA language support, such as pure Python projects. Using `SetupCUDA.cmake` to set HoloHub project defaults would add an unnecessary dependency in those cases.
2. Most HoloHub projects do not consider portability at all, but instead demonstrating building and running an application on only the user's machine. `SetupCUDA.cmake` tailors a portable list of architectures for CUDA compilation and is mostly extraneous in that case.

Based on that, I suggest that `"native"` might be the appropriate `CMAKE_CUDA_ARCHITECTURES` _default_ value in HoloHub. Reflecting the details above:
1. Setting the CMake `CMAKE_CUDA_ARCHITECTURES` variable at the top level _without_ enabling the CUDA language makes the default available to projects using CUDA without introducing a new dependency to non-CUDA projects.
2. Setting the default as "native" matches the standard HoloHub default target platform which is the user's system. Projects requiring portability can override that default such as done in `SetupCUDA.cmake`.

@AndreasHeumann what do you think?